# Running the Workflows

## NMDC EDGE QuickStart

<div align="center">
    <video src="https://nmdc-edge.org/docs/videos/nmdc-edge.mp4" width="80%" height="80%" controls />
</div>
<br><br>

> ### NMDC EDGE QuickStart Tutorial Practice
>
>Task 1:  Create an NMDC EDGE account with either your email address or your ORCiD account.
>
>Task 2: Download the small interleaved data file listed here. Upload the file to NMDC EDGE.
>
>Task 3: Click the user icon (in the top right corner with your initials) and under “Files”, click on “Manage Uploads”. Verify that the file you uploaded is there. (Note: Later you can delete uploads that are no longer needed.)
>
>Task 4 (optional):  Click the user icon and under “Account”, click on “Profile”. Edit your account to receive email notification of project status by clicking “ON”.

## Metagenomics

### ReadsQC

<div align="center">
    <video src="https://nmdc-edge.org/docs/videos/ReadsQC.mp4" width="80%" height="80%" controls />
</div>
<br><br>

>NMDC EDGE ReadsQC Tutorial Practice 
>
>Task 1: Log into NMDC EDGE and run the ReadsQC workflow using the dataset uploaded in the QuickStart tutorial.
>
>Task 2:  How many reads were in the input file? How many bases were in the input file?
>
>Task 3:  How many reads were in the output file? How many bases were in the output file?
>
>Task 4:  What file in the output would be used in the next workflow?


### Read-based Taxonomy Classification

<div align="center">
    <video src="https://nmdc-edge.org/docs/videos/ReadBasedAnalysis.mp4" width="80%" height="80%" controls />
</div>
<br><br>

>NMDC EDGE Read-based Taxonomy Classification Tutorial Practice 
>
>Task 1: Log into NMDC EDGE and run the Read-based Taxonomy Classification workflow. Select all three Analysis tools.  Using the clean data output file from project run in the ReadsQC Tutorial (the file ending in .anqdpht.fq.gz). In this case, the file will be treated as single-end reads.
>
>Task 2:  How many of the Top 10 species are called by more than one tool?
>
>Task 3:  List the genera that are called by all three tools in the Top 10.
>
>Task 4:  Open a full window of the Krona plot at species level. What percentage of the sample is estimated to be Pseudomonas aeruginosa? Take a screenshot of the Krona plot.

### Assembly

<div align="center">
    <video src="https://nmdc-edge.org/docs/videos/MetagenomeAssembly.mp4" width="80%" height="80%" controls />
</div>
<br><br>

### Annotation

<div align="center">
    <video src="https://nmdc-edge.org/docs/videos/MetagenomeAnnotation.mp4" width="80%" height="80%" controls />
</div>
<br><br>

### MAGs Generation

<div align="center">
    <video src="https://nmdc-edge.org/docs/videos/MetagenomeMAGs.mp4" width="80%" height="80%" controls />
</div>
<br><br>

### Running multiple workflows or the full metagenomic pipeline with a single input

<div align="center">
    <video src="https://nmdc-edge.org/docs/videos/pipeline.mp4" width="80%" height="80%" controls />
</div>
<br><br>
