#  Running the Workflows

## Metagenomics

### QuickStart
#### Register for an account 
Users must register for an account within the NMDC EDGE platform or login using the user’s ORCiD account.  
![1_Login1](https://user-images.githubusercontent.com/10235220/156612523-aba07096-a5c6-42fd-a972-8312fb138fcd.jpeg)
![2_Login2](https://user-images.githubusercontent.com/10235220/156612527-87c16b4c-1c42-4d11-9995-a3036ed81fe3.jpeg)
#### User Profile
Once logged in, the green button with the user’s initials on the right provides a drop-down menu which allows the user to manage their projects and uploads; there is also a button which allows users to edit their profile. On this profile page, there are two options: 1) the option to receive email notification of a project’s status (OFF by default) and 2) the option to change the user’s password (also OFF by default).
![3_User_Profile](https://user-images.githubusercontent.com/10235220/156614511-69517d6f-6b67-452a-a4fb-f0fe02fbce22.jpeg)
#### Upload data
Two options are available for users to upload their own data to process through the workflows. The first is using the button in the left menu bar. The second is through the drop-down menu shown when clicking the green button with the user’s initials on the right. Either button will open a window which allows the user to drag and drop files or browse for the user’s data files. (There are also some datasets in the Public Data folder for users to test the platform.) 
![4_Upload_Data](https://user-images.githubusercontent.com/10235220/156614797-aebee17d-c738-43c9-818e-09a9cb2a958e.jpeg)
#### Running a single workflow 
To run a workflow, the user must provide:
1.	A unique Project/Run Name with no spaces (underscores are fine).
2.	A description is optional, but recommended.
3.	The user then selects the workflow desired from the drop-down menu.
4.	For metagenomic/metatranscriptomic data, the user must also select if the input data is interleaved or separate files for the paired reads.
5.	Then the input file(s) from the available list of files. 
6.	The user should click “Submit.
![5_Single_workflow_1](https://user-images.githubusercontent.com/10235220/156614921-a1ab271b-6575-401c-b095-c06c455a6c72.jpeg)
Note: Clicking on the buttons to the right of the data input blanks opens a box called “Select a file” to allow the user 
to find the desired files (shown in purple) from previously run projects, the public data folder, or user uploaded files. 
![6_Single_workflow_input](https://user-images.githubusercontent.com/10235220/156614996-b63729be-1eee-493e-a01c-4ea4013aa091.jpeg)
#### Running multiple workflows
1.	Another option is to select “Run Multiple Workflows” if the user desires to run more than one of the metagenomic workflows or the entire metagenomic pipeline. 
2.	Enter a unique Project/Run Name with no spaces (underscores are fine).
3.	A description is optional, but recommended.
4.	The user must also select if the input data is interleaved or separate files for the paired reads.
![7_Multiple_workflows_1](https://user-images.githubusercontent.com/10235220/156615117-4d691332-1573-4cd8-a26e-40b4d0852f06.jpeg)
![8_Multiple_workflows_2](https://user-images.githubusercontent.com/10235220/156615201-0cac2cfa-187e-4e95-adcd-ae8a14158a6d.jpeg)


### ReadsQC


### Read-based Taxonomy Classification

### Assembly

### Annotation

### MAGs Generation

### Running multiple workflows or the full metagenomic pipeline with a single input
